### DISCUSSION TITLE
How i set my consts with value from setValues?

### DISCUSSION THREAD

**Discussion by PereiraDiegoMiguel** [react-hook-form/react-hook-form#3103] [Q&A]
const [bank, setBank] = useState();
  const [agency, setAgency] = useState();

  const { register, handleSubmit, errors, setValue } = useForm<IFormInputs>({
    resolver: yupResolver(validations),
  });

  const onSubmit = (data: IFormInputs) => {
    return (
       //It doesn't work:
       setBank(data.bank),
       setAgency(data.agency),
    )
}
useEffect(() => {
    register({ name: 'bank' }, { required: true });
    register({ name: 'agency' }, { required: true });
  }, [register]);

 return (
    <>
            <Form
              ref={formRef}
              onSubmit={handleSubmit(onSubmit)>

              <Input
                ref={bankRef}
                name='bank'
                autoCapitalize='words'
                icon='lock'
                placeholder='Banco'
                returnKeyType='next'
                value={bank}
                onSubmitEditing={() => { agencyRef.current?.focus(); }}
                onChangeText={value => setValue('bank', value)}
              />
                <Input
                  ref={agencyRef}
                  keyboardType='number-pad'
                  name='agency'
                  icon='lock'
                  placeholder='Agência'
                  returnKeyType='next'
                  value={agency}
                  onSubmitEditing={() => { dgAgencyRef.current?.focus(); }}
                  onChangeText={value => setValue('agency', value)}
                />

              <Button
                title='Enviar'
                color='rgb(219, 28, 35), rgb(233, 26, 68))'
                onPress={handleSubmit(onSubmit)}
              />
            </Form>

**Comment by bluebill1049** [comment]
csb and steps to reproduce.

**Reply by PereiraDiegoMiguel** [reply]
const onSubmit = (data: IFormInputs) => {
return (

// It doesn't work:

setBank(data.bank),
setAgency(data.agency),
)

**Reply by bluebill1049** [reply]
you are using customer register, which is register at useEffect, none of the ref is been registered, i would recommend you to have a read on the controller section: https://react-hook-form.com/api#Controller and use controller instead.

**Reply by PereiraDiegoMiguel** [reply]
estava perguntando se tinha uma maneira mais facil de pegar o somente o valor do "setvalue", ao invés de pegar o campo e o valor. Mas pelo que você falou, somente se usar o Controller, correto ?

O ref estava utilizando para saltar os campos apóso o usuário o clicle do usuário.


Do you have another solution or just with the controller?

**Reply by PereiraDiegoMiguel** [reply]
I was asking if there was an easier way to get only the value of the "setvalue", when suggesting to take the field and the value. But from what you said, only if you use the Controller, correct?
